### PR TITLE
django 2.0 support; drop django 1.10 support

### PR DESCRIPTION
**Pull request by willkg**
This drops support for Django 1.10 (#202).

This does a first pass at adding support for Django 2.0 (#201).

Figured I'd do them both at the same time since tox was hosed in the interim, so PRs wouldn't pass CI.

I don't use Django 2.0 on anything, yet, so while these changes work with tox, I don't know if they completely fix Django 2.0 support issues. But they do fix CI so then we can continue doing development.

### CODE REVIEW

**Comment by peterbe** [comment]
I haven't done a review yet, but I can attest that it makes things work. 
In https://github.com/mozilla-services/tecken/pull/625 I upgraded Tecken to Django 2.0 and the only thing I couldn't get to work was the mozilla-django-oidc imports. 
So I did...
```
root@42e6cec5daac:/app# pip install -I git+git://github.com/willkg/mozilla-django-oidc.git@201-django-20#egg=mozilla_django_oidc
```
and then all my tests passed! 🎊

Note. I haven't actually done much testing with https://github.com/mozilla-services/tecken/pull/625 yes. Just wanted the tests to work.

File: tox.ini [hunk 1]
```diff
@@ -2,8 +2,8 @@
 envlist =
     lint
     py{27,33,34,35}-django18
-    py{27,34,35}-django110
     py{27,34,35,36}-django111
+    py{34,35,36}-django200
```
Comment (peterbe) [inline tox.ini]: +1

**Comment by willkg** [comment]
Can someone land this or is there something else that should get done here?

**Comment by johngian** [comment]
Sorry for the late response (i was on vacation for some time).
I tested it manually here [1] using a Circle CI workflow for E2E testing i am currently working on [2] (still WIP).

It looks all supported Django versions are working with both Python 2,3 and HS,RS authentication algorithms.

Code looks OK. r+ 

[1] https://circleci.com/workflow-run/f1d2c4ed-c6e6-4470-9cfb-96857de1fd48
[2] https://github.com/mozilla/mozilla-django-oidc/pull/205

**Comment by willkg** [comment]
Thank you!

**Comment by peterbe** [comment]
I've tested it in Tecken and it's working as fine as I can possibly see.
https://github.com/mozilla-services/tecken/pull/625

**Pull request merged**